crates/ra_hir/src/nameres/collector.rs

```diff
@@ -295,6 +299,21 @@ where
                         }
                     }
 
+                    // `#[macro_use] extern crate` glob imports all macros exported,
+                    // ignoring their scopes
+                    if import.is_extern_crate && import.is_macro_use {
+                        if let Some(ModuleDef::Module(m)) =
+                            def.a().and_then(|item| item.take_types())
+                        {
+                            tested_by!(macro_rules_from_other_crates_are_visible_with_macro_use);
+
+                            let item_map = self.db.crate_def_map(m.krate);
+                            for (name, &macro_id) in &item_map.exported_macros {
+                                self.define_macro(module_id, name.clone(), macro_id, false);
```

Comment:
I think this shouldn't call `define_macro`, and instead shoult just directly add the macro to the global scope, bypassing the module scope.

That is, the following code fails with unresovled macro error:

```
#[macro_use]
extern crate lazy_static;

crate::lazy_static! {
    static ref XS: u32 = 92;
}

fn main() {
    println!("Hello, world!");
}
```